hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RSProcedureDispatcher.java

```diff
@@ -330,6 +322,73 @@ private boolean scheduleForRetry(IOException e) {
       return true;
     }
 
+    private boolean unableToConnectToServer(IOException e) {
+      // This exception is thrown in the rpc framework, where we can make sure that the call has not
+      // been executed yet, so it is safe to mark it as fail. Especially for open a region, we'd
+      // better choose another region server.
+      // Notice that, it is safe to quit only if this is the first time we send request to region
+      // server. Maybe the region server has accepted our request the first time, and then there is
+      // a network error which prevents we receive the response, and the second time we hit a
+      // CallQueueTooBigException, obviously it is not safe to quit here, otherwise it may lead to a
+      // double assign...
+      if (e instanceof CallQueueTooBigException) {
+        LOG.warn("request to {} failed due to {}, try={}, this usually because"
+          + " server is overloaded, give up", serverName, e, numberOfAttemptsSoFar);
+        return true;
+      }
+      if (isSaslError(e)) {
+        LOG.warn("{} is not reachable; give up after first attempt", serverName, e);
+        return true;
+      }
+      return false;
+    }
+
+    private boolean isSaslError(IOException e) {
+      if (
+        e instanceof SaslException || (e.getMessage() != null
+          && e.getMessage().contains(RpcConnectionConstants.RELOGIN_IS_IN_PROGRESS))
```

Comment:
yes, we have come across this as well

Reply:
`procedure.RSProcedureDispatcher - request to rs-2,61020,1692796686773 failed due to java.io.IOException: Call to address=rs-2:61020 failed on local exception: java.io.IOException: Can not send request because relogin is in progress., try=0, retrying..`

Reply:
OK, I misunderstood the logic here. So here even if it is not a SaslException but the log message is 'relogin in progress', we will also consider it as a SaslExcception.